File: toy/sns/src/main/java/me/kzv/sns/controller/response/UserJoinResponse.java
package me.kzv.sns.controller.response;

import lombok.AllArgsConstructor;
import lombok.Getter;
import me.kzv.sns.model.User;

@Getter
@AllArgsConstructor
public
class UserJoinResponse {
    private Integer id;
    private String name;

    public static UserJoinResponse fromUser(User user) {
        return new UserJoinResponse(
                user.getId(),
                user.getUsername()
        );
    }

}

File: toy/simple-sns/src/main/kotlin/me/kzv/simplesns/controller/response/UserJoinResponse.kt
package me.kzv.simplesns.controller.response

import me.kzv.simplesns.model.User
import me.kzv.simplesns.model.enum.UserRole

class UserJoinResponse(
    val userName: String,
    val id: Long?,
    val role: UserRole,
) {
    companion object {
        fun fromUser(user: User) = UserJoinResponse(
            userName = user.userName,
            id = user.id,
            role = user.userRole,
        )
    }
}